rpc/server_auth.go

```diff
@@ -171,6 +177,9 @@ func (ss *simpleServer) authUnaryInterceptor(
 	if !ss.exemptMethods[info.FullMethod] {
 		nextCtx, err := ss.ensureAuthed(ctx)
 		if err != nil {
+			if status, _ := status.FromError(err); status.Code() == codes.Unauthenticated && !ss.canByPassAuthCheck(info.FullMethod) {
```

Comment:
```suggestion
			if status, _ := status.FromError(err); status.Code() == codes.Unauthenticated && ss.canByPassAuthCheck(info.FullMethod) {
```

Shouldn't this be run if we `can` bypass auth checks rather than if we cant bypass auth checks

Reply:
my bad! I originally made ss. canByPassAuthCheck ss.requiresAuthCheck so forgot to remove the negation